chef/cookbooks/cinder/recipes/volume.rb

```diff
@@ -88,8 +88,16 @@ def make_volume(node, backend_id, volume)
 
   return if volume_exists(volname)
 
-  unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node)
-  claimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.claimed(node, "Cinder")
+  claim_string = "Cinder"
+
+  # do not claim any disk that has non-Cinder role assigned
+  disk_roles = node["crowbar_wall"].disk_roles || {}
+  unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node).sort.select { |d|
+    role = disk_roles[d.unique_name] || ""
+    role.empty? || role == claim_string
```

Comment:
@jsuchome but it does make sense to add some helper in the barclamp library to do what you're doing here (and in the other barclamps that use this disk role stuff), doesn't it?